db/trie/context.go

```diff
@@ -0,0 +1,133 @@
+// Copyright (c) 2018 IoTeX
+// This is an alpha (internal) release and is not suitable for production. This source code is provided 'as is' and no
+// warranties are given as to title or non-infringement, merchantability or fitness for purpose and, to the extent
+// permitted by law, all liability for your use of the code is disclaimed. This source code is governed by Apache
+// License 2.0 that can be found in the LICENSE file.
+
+package trie
+
+import (
+	"context"
+
+	"github.com/golang/protobuf/proto"
+	"github.com/pkg/errors"
+
+	"github.com/iotexproject/iotex-core/db"
+	"github.com/iotexproject/iotex-core/pkg/hash"
+	iproto "github.com/iotexproject/iotex-core/proto"
+)
+
+type sameKeyLenTrieContextKey struct{}
+
+// SameKeyLenTrieContext provides the trie operation with auxiliary information.
+type SameKeyLenTrieContext struct {
+	// Bucket defines the bucket to store in db
+	Bucket string
+	// DB refers to the db to store the key-value pairs
+	DB db.KVStore
+	// CB is a collection of db operations
+	CB db.CachedBatch
+	// KeyLength defines the length of the key to be used in the trie
+	KeyLength int
+	// InitRootHash stores the root hash
+	InitRootHash hash.Hash32B
```

Comment:
i feel branchRootTrie struct{} is a better place to put rootHash, b/c rootHash is more closely related to branchRootTrie, and constantly changing. All other vars in context are only set at init time, and keep the same later

also by moving it we can do tr.RootHash instead of tr.tc.RootHash in branchRootTrie.go